galaxy-cli tasking polling interval from environment variable

jctanner opened the pull request:
##### SUMMARY

2 seconds is a long time relatively speaking for a battery of integration tests that call galaxy cli over and over.

##### ISSUE TYPE

- Feature Pull Request

##### ADDITIONAL INFORMATION

This is probably better done in the true ansible config structure and or via a cli arg, but I needed something right now.

jctanner committed:
Allow an environment variable to override the galaxy server task polling time.

No-Issue

Signed-off-by: James Tanner <tanner.jc@gmail.com>

bcoca reviewed (changes requested):
Create a configuration entry, that will also have the benefit of documenting it

s-hertel committed:
Add a configuration entry and changelog

jborean93 commented on lib/ansible/config/base.yml:
```diff
@@ -1528,6 +1528,14 @@ GALAXY_REQUIRED_VALID_SIGNATURE_COUNT:
+  type: float
+  default: "2.0"
+  env:
+    - name: ANSIBLE_GALAXY_SLEEP_SECONDS_POLLING
```
```suggestion
    - name: ANSIBLE_GALAXY_SLEEP_SECONDS_POLLING
  version_added: '2.18'
```

We should say when the option was added.

s-hertel replied:
Thanks, good catch.

s-hertel committed:
Update lib/ansible/config/base.yml

Co-authored-by: Jordan Borean <jborean93@gmail.com>

jborean93 commented on changelogs/fragments/83803-collection-import-poll-interval.yml:
```diff
@@ -0,0 +1,4 @@
+minor_changes:
+- >-
+  ``ansible-galaxy collection import`` - add a configuration option for the initial poll interval
```
Sorry didn't pick this up, but I believe the command that calls this code is `publish`. The `import` one is for roles.

```suggestion
  ``ansible-galaxy collection publish`` - add a configuration option for the initial poll interval
```

s-hertel committed:
fix changelog

s-hertel commented:
@jctanner Do you think the hardcoded exponential factor (1.5) needs to be configurable too?

jctanner commented:
@s-hertel oh hey, sorry i hadn't even seen the notifcations on this. Thanks for patching it!

I'm not sure about the "hardcoded exponential factor" but maybe it would be good to have that configurable too.

jborean93 commented:
> I'm not sure about the "hardcoded exponential factor" but maybe it would be good to have that configurable too.

The wait is multiplied by the factor after every interval, so it'll wait for 2 seconds, then 3, 4.5, until it hits the max of 30 seconds.

bcoca commented on lib/ansible/config/base.yml:
```diff
@@ -1528,6 +1528,15 @@ GALAXY_REQUIRED_VALID_SIGNATURE_COUNT:
+    - The initial interval in seconds for polling the import status of a collection, using exponential backoff with a factor of 1.5.
+    - The maximum total backoff is capped at 30 seconds.
+  type: float
+  default: "2.0"
```
if you remove the quotes it should avoid conversion

s-hertel replied:
Won't it be converted to an int due to YAML?

jborean93 replied:
AFAIK It'll be a float because of the `.0`.

s-hertel committed:
add a configuration option for the exponential factor

rename the environment variable for clarity, just in case future configuration options are added for roles or other ansible-galaxy subcommands

s-hertel committed:
remove quotes
